src/Xamarin.Android.Build.Tasks/MSBuild/Xamarin/Android/Xamarin.Android.Bindings.Maven.targets

```diff
@@ -0,0 +1,35 @@
+<!--
+***********************************************************************************************
+Xamarin.Android.Bindings.Maven.targets
+
+This file contains MSBuild targets used to enable @(AndroidMavenLibrary) support.
+
+***********************************************************************************************
+-->
+
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <UsingTask TaskName="Xamarin.Android.Tasks.MavenDownloadTask" AssemblyFile="Xamarin.Android.Build.Tasks.dll" />
+
+  <PropertyGroup>
+    <!-- Maven cache directory, locations copied from XBD which should be sufficiently battle-tested -->
+    <MavenCacheDirectory Condition="'$(OS)'=='Unix' and '$(MavenCacheDirectory)'==''">$(HOME)\Library\Caches\MavenCacheDirectory\</MavenCacheDirectory>
```

Comment:
Is this specified/controlled by MavenNet, or is this something *we* control?  If it's something we control, should we instead use `Library/Caches/dotnet-android`?

Reply:
This is a location of our choosing.  I feel like it should denote in some way that this contains Maven artifacts.  We could go deeper if we also want to denote it comes from `dotnet-android`, like:
```
Library/Caches/dotnet-android/MavenCacheDirectory
```

Another data point to consider is what NuGet uses:
- Windows: `%userprofile%\.nuget\packages`
- Mac/Linux: `~/.nuget/packages`

Reply:
Too deep folder structure might cause issues on Windows. Do not forget that we need unpacking too.

Reply:
Not sure how to make this work in an MSBuild context, but I think that a subdirectory of [`Environment.SpecialFolder`](https://learn.microsoft.com/en-us/dotnet/api/system.environment.specialfolder?view=net-7.0)`. InternetCache` might be best:

> The directory that serves as a common repository for temporary Internet files.

For [macOS this is `$HOME/Llibrary/Caches`](https://github.com/dotnet/runtime/blob/97f773875851aa7077d208236f91da69b1c31011/src/libraries/System.Private.CoreLib/src/System/Environment.GetFolderPathCore.Unix.cs#L117).

For [Windows this is `Interop.Shell32.KnownFolders.InternetCache`](https://github.com/dotnet/runtime/blob/97f773875851aa7077d208236f91da69b1c31011/src/libraries/System.Private.CoreLib/src/System/Environment.Win32.cs#L253) which is [`CSIDL_INTERNET_CACHE`](https://learn.microsoft.com/en-us/windows/win32/shell/csidl) which is:

> file system directory that serves as a common repository for temporary Internet files. A typical path is C:\Documents and Settings\username\Local Settings\Temporary Internet Files.

(which just reads like Vista, and presumably has changed since Windows 7…)

[Looks like you should be able to use MSBuild property functions](https://stackoverflow.com/a/72913055), a'la:

```xml
    <MavenCacheDirectory Condition=" '$(MavenCacheDirectory)' == '' "
      >$([System.IO.Path]::Combine($([System.Environment]::GetFolderPath(SpecialFolder.InternetCache)), "dotnet-android", "Maven"))</MavenCacheDirectory>
```

Reply:
It looks like on Windows 11, this is:
```
C:\Users\jopobst\AppData\Local\Microsoft\Windows\INetCache
```

I don't like that this is hidden underneath `Microsoft\Windows`. If I'm looking for a dotnet-android Maven cache directory I wouldn't be checking there.

It also appears to be largely unused, my folder has a few directories totaling less than 1MB.

Sounds like what we want is:

Mac/Linux: `($HOME)/Library/Caches/dotnet-android/MavenCacheDirectory/`
Windows: `$(LocalAppData)\dotnet-android\MavenCacheDirectory\`

I will update to these values.

Reply:
@jpobst [wrote](https://github.com/xamarin/xamarin-android/pull/8420#discussion_r1377931744):

> Sounds like what we want is:
> 
> Mac/Linux: ($HOME)/Library/Caches/dotnet-android/MavenCacheDirectory/

Not quite; `$(HOME)/Library` is a macOS-ism, *not* a Unix-ism.

The closest Linux-ism would likely be [`$(XDG_CACHE_HOME)`](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html), which defaults to `$(HOME)/.cache`.

@grendello: what is `System.Environment.GetFolderPath(System.Environment.SpecialFolder.InternetCache))` on Linux?  [Looks like it should be `string.Empty`](https://github.com/dotnet/runtime/blob/97f773875851aa7077d208236f91da69b1c31011/src/libraries/System.Private.CoreLib/src/System/Environment.GetFolderPathCore.Unix.cs#L147), which isn't helpful.

@jpobst: perhaps instead of using `$(OS)` we need to use the MSBuild OS property functions, using [`OperatingSystem.IsOsPlatform(string)`](https://learn.microsoft.com/en-us/dotnet/api/system.operatingsystem.isosplatform?view=net-7.0), with `Windows` first, `macOS` second, and "Unix" as the fallback?